### PR TITLE
adjust hifiasm for small inputs

### PR DESCRIPTION
**Pull request by bgruening**

### CODE REVIEW

**Comment by bgruening** [comment]
@cat-bro does this look good for you?

**Comment by cat-bro** [comment]
This is the first file size rule in the shared database. The implication of this rule is that if anyone wanted to have 10 cores and 50 mem for a 0.05 input size they would need to have a rule for hifiasm in their own config explicitly covering that file size.

This rule needs an id so it can be easier to override, such as 'small_job_rule', 'small_file_size_rule'.

@nuwang I remember us talking about this prior to GCC.  I'm wondering if we should be generalising this behaviour in the default tool here, such as

```
  default_tool:
    context:
       small_job_max_file_size: null
       small_job_cores: null
       small_job_mem: null
 ...
 rules:
     - id: small_job_rule
        if: small_job_max_file_size and small_job_cores and small_job_mem and input_size < small_job_file_size:
        cores: small_job_cores
        mem: small_job_mem
```

Of course this does also result in downstream galaxies inheriting the change but it could make it easier to work with.

If not the above, it would be good to have a convention for the rule ID for this type of rule, e.g. 'small_job_rule' or 'tpv_small_job_rule'

**Comment by bgruening** [comment]
> This rule needs an id so it can be easier to override, such as 'small_job_rule', 'small_file_size_rule'.

Why is this? Why can I not just have a hifiasm rule in my local config? I thought that we add here our best-guess rules and everyone that disagrees adds something like the following to the local rules:

```
  toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/.*:
    cores: 40
    mem: 220

**Comment by cat-bro** [comment]
Having IDs for rules is a good convention anyway because it helps with debugging.

In this case, if Galaxy Tasmania has 
```
  toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/.*:
    cores: 40
    mem: 220
```

and uses the shared db they will still inherit any rules associated with hifiasm in the shared db.  If the rule has an ID, that just makes it a little easier for Galaxy Tasmania to change the parameters of the rule they have need to.

My example above with context values probably doesn't work because there could be any number of settings in a rule, not just cores and mem.

**Comment by nuwang** [comment]
I do recall that conversation, but I can no longer recall whether there's a benefit to introducing these variables. It seems simpler, at least in this case to add an id only:
```
    rules:
    - if: input_size <= 0.2
      id: small_input_rule
      cores: 6
      mem: 40
```
And downstream users can trivially override this rule. I suppose there could be benefits for very complex rules, which one may not want to repeat downstream. For those cases, perhaps we can introduce overridable context variables on a case-by-case basis?

File: tools.yml [hunk 1]
```diff
@@ -185,6 +185,10 @@ tools:
   toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/.*:
     cores: 20
     mem: 120
+    rules:
+    - if: input_size <= 0.2
```
Comment (nuwang) [inline tools.yml:189]:
```suggestion
    - if: input_size <= 0.2
      id: hifiasm_small_input_rule
```

File: tools.yml [hunk 2]
```diff
@@ -185,6 +185,11 @@ tools:
...
+      id: hifiasm_small_input_rule
```
Comment (nuwang) [inline tools.yml:190]: tpv lint fix
```suggestion
    - id: hifiasm_small_input_rule
      if: input_size <= 0.2
```

**Comment by nuwang** [comment]
Thanks @bgruening for the first file size rule!

**Comment by bgruening** [comment]
Thanks lets add many more :)

**Pull request merged**